Soe soeop 368

mdyoung3 opened the pull request:
# READY FOR REVIEW
Ready for review.

# Summary
- Docksal configuration scripts that set up local docker containers and build the site. 

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- 3/10
- Affects SOE local setup.

# Review Tasks

## Setup tasks and/or behavior to test

1. Clone site locally
2. `Cd` into project root folder
3. Read the README.md file located in .`docroot` folder for steps to set up site locally.

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SOEOP-368

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

sherakama committed:
Update settings.php

jbickar committed:
SOEOP-000: Updated google_tag-7.x-1.6

mdyoung3 committed:
Added files folder to gitignore.

mdyoung3 committed:
SOE-SOEOP-368 - Added Docksal scripts to install necessary Grunt tool for scaffolding this site. See README.MD in .docksal folder for instructions.

mdyoung3 committed:
SOEOP - fixed typo in README.

Commit:
SOE-SOEOP-368 - replaced fin command with bash command.

Commit:
SOE-SOEOP-368 - updated message; removed site-init and corrected path to root for aliases

Commit:
SOE-SOEOP-368 - added script to get site ready for development and fix file permissions issue..

jbickar committed:
docksal config and Github templates (#19)

* fixup! docksal documentation and scripts

* fixup! Add .github templates

* SOEOP-368: Additional docksal config

* fixup! README

* fixup! WIP

* fixup! local env

jbickar reviewed (changes requested):
This PR should be submitted to the `soe-develop-mc` branch.

mdyoung3 commented:
@jbickar Switched base branch for PR.

jbickar commented on .docksal/README.md:
````diff
@@ -0,0 +1,49 @@
+4. If you use a key other than `id_rsa` to connect to Acquia, run `fin ssh-key add <keyname>` (e.g., `fin ssh-key add id_rsa_acquia_4096`)
+5. Run `fin init`
+
+
+Building Notes:
+`npm install` failed. Trying to downgrade. Downgrading success. Using node -v 10.22.x
+
+Grunt failed when I configured Gruntconfig.json file for drush in /vendor folder. To find correct location, had to use `whereis` command.
+
+Be sure to add path to src. Refer to the following:
+
+```
+"srcPaths": {
+           "make": "src/engineering.make",
+           "drupal": "src"
+         },
+```
+
+Need aliases from Acquia
+
+Running `grunt settings` results in an error that the same files are already in the folder. Skipping step.
+
+##Links
+https://github.com/SU-SOE/grunt-drupal-tasks/blob/SOE-develop-mc/docs/soe_build_notes.md
+https://stackoverflow.com/questions/47008159/how-to-downgrade-node-version
+https://www.npmjs.com/package/n
+https://phase2.github.io/grunt-drupal-tasks/10_BUILD/
+https://www.drupal.org/node/1401522
+
+
````
I believe we discussed dropping these, as they are internal notes.

mdyoung3 replied:
Yes, however, I'm reluctant to remove them. They contain some research regarding the setting up Grunt for this project. (E.g., that it will only work with an older version of Node.)

I'd recommend transferring relevant information therein to the Build Notes. Re: https://github.com/SU-SOE/grunt-drupal-tasks/blob/SOE-develop-mc/docs/soe_build_notes.md

jbickar commented on .docksal/src_files/Acquia_settings/.acquia/cloudapi.conf:
```diff
@@ -0,0 +1 @@
+{"email":"mdyoung3@stanford.edu","key":"gOoBtv57btCoLt0EXa8yzmas\/ESORvy2dV1LVQ10W+V7TdCyVQuUR28GgvMeQV9bH5zFOuKqsjP4"}
```
Probably don't want to commit an API key to a public repo. And you now probably should revoke that and re-generate.

mdyoung3 replied:
Agreed. I'll remove it. 

Probably have to update the README to include instructions for setting up AC tokens.

jbickar commented on .docksal/src_files/Acquia_settings/.drush/cloudapi.acquia.com.pem:
```diff
@@ -0,0 +1,99 @@
+-----BEGIN CERTIFICATE-----
```
Why do we need the certificate?

jbickar commented on .docksal/src_files/Acquia_settings/.drush/hsmrc.aliases.drushrc.php:
```diff
@@ -0,0 +1,82 @@
+<?php
```
I would only commit the `swssoe` drush aliases for Acquia.

jbickar commented on .docksal/src_files/Acquia_settings/.drush/stanfordfsh.aliases.drushrc.php:
```diff
@@ -0,0 +1,82 @@
+<?php
```
I would only commit the `swssoe` drush aliases for Acquia.

jbickar commented on .docksal/src_files/Acquia_settings/.drush/swssoe.aliases.drushrc.php:
```diff
@@ -0,0 +1,122 @@
+<?php
```
Yes, these. I won't leave comments on each of the other alias files. 😄

mdyoung3 replied:
Sounds good.

jbickar commented on .docksal/src_files/Acquia_settings/README-acquiacloud.txt:
```diff
@@ -0,0 +1,3 @@
+This folder contains two hidden folders named .acquia and .drush. Both of these folders should be placed in the $HOME directory on your system. The .acquia folder contains your Acquia Cloud API credentials and the .drush folder contains your Drush aliases for the sites you currently have access to on Acquia Cloud.
```
I don't think we need this file.

jbickar commented on .docksal/src_files/Gruntfile.js:
```diff
@@ -0,0 +1,18 @@
+  require('grunt-drupal-tasks')(grunt);
+  grunt.loadNpmTasks('grunt-deploy-site');
+  grunt.loadNpmTasks('grunt-symlink');
+  grunt.config.set('deploy_site.development.options.branch', 'SOE-SOEOP-368');
```
The default here probably should be `soe-develop-mc`. Future improvement would be to take this as an argument or from a variable.

jbickar commented on .docksal/commands/dev-mode:
```diff
@@ -0,0 +1,16 @@
+drush variable-set preprocess_css 0
+
+find "$DOCROOT_PATH/sites/default/files" -type f -exec sudo chmod 644 {} \;
+find "$DOCROOT_PATH/sites/default/files" -type d -exec sudo chmod 755 {} \;
```
Please add a blank line to the end of each bash file.

mdyoung3 replied:
Done

sherakama replied:
You should be able to configure your IDE to do this automagically.

jbickar reviewed (changes requested):
Please review comments and make the associated changes.

jbickar committed:
fixup! .acquia directory

jbickar commented on .docksal/commands/grunt-init:
```diff
@@ -0,0 +1,51 @@
+rsync -az "$PROJECT_ROOT/.docksal/src_files/settings.local.php" "$PROJECT_ROOT/soe/src/sites/default"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/Acquia_settings/.drush/"* "$HOME/.drush"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/soe.aliases.drushrc.php"* "$HOME/.drush"
+rsync -az -r "$PROJECT_ROOT/.docksal/src_files/Acquia_settings/.acquia" "$HOME"
```
This line now can be removed as there's nothing in there.

jbickar commented on .docksal/README.md:
```diff
@@ -0,0 +1,49 @@
+5. Run `fin init`
+
+
+Building Notes:
```
```suggestion
## Building/Troubleshooting Notes:
```

jbickar replied:
Per discussion, move this below the `## Links` section, below.

jbickar commented on .docksal/README.md:
````diff
@@ -0,0 +1,49 @@
+
+Grunt failed when I configured Gruntconfig.json file for drush in /vendor folder. To find correct location, had to use `whereis` command.
+
+Be sure to add path to src. Refer to the following:
+
+```
+"srcPaths": {
+           "make": "src/engineering.make",
+           "drupal": "src"
+         },
+```
+
````
Per discussion, drop this from the README for Docksal, and submit a separate PR to add this change to the `soe/Gruntconfig.json` file

jbickar committed:
SOEOP-000: Delete and ignore behat.local.yml

jbickar committed:
fixup! gitignore

jbickar committed:
Merge branch 'jbickar-patch-1' of github.com:SU-SOE/grunt-drupal-tasks into jbickar-patch-1

jbickar committed:
Merge branch 'jbickar-patch-1' into SOE-SOEOP-368

Commit:
Fixup! SOEOP-368 - Johns feedback. See notes for some update information.

--Removed superfluous drush alises
--Added behat local file
--Turned off HTTPS
--Included

mdyoung3 commented:
Hi @jbickar 

I pushed up the fixes with the changes/removals/edits per your feedback.

jbickar commented:
@mdyoung3, is this ready for review again? Can you resolve the `.gitignore` conflict if so?

We should get this in so that I can review #24 (I'm reviewing it, it's just that I'm working off of multiple branches).

Commit:
Merge branch 'SOE-develop-mc' into SOE-SOEOP-368

mdyoung3 commented:
HI @jbickar 

Sorry I missed your message. The gitignore is merged and it is ready for review. 

Best,
Marc Young

jbickar reviewed (approved):
GTG
